beta/index.markdown

```diff
@@ -0,0 +1,183 @@
+each of their fellow contributors, so 9 patches, they will be facing _9 hours_
+before they know if any of these patches interact negatively with any of the
+others. This compounds as failures happen. If the second patch to land introduces
+an API breakage in which its tests pass, but many other new changes that rely
+on the old behavior break, suddenly you may only land 1 patch in 9 hours.
+
+While the [JClouds plugin][jc] helps with pre-approval checking parallelism,
+it can do nothing with this problem.
+
+This is why [Zuul][z] implements speculative futures. In the 9 patch scenario above,
+[Zuul][z] will spin up _9_ cloud nodes all at once. It will try to run a job for
+each potential future in the master branch. So if the patches are numbered, it will
+run each of these futures in parallel:
+
+
+    A 1
+    B 1+2
+    C 1+2+3
+    D 1+2+3+4
+    E 1+2+3+4+5
+    F 1+2+3+4+5+6
+    G 1+2+3+4+5+6+7
+    H 1+2+3+4+5+6+7+8
+    I 1+2+3+4+5+6+7+8+9
+
+If patch #2 introduces an API break that breaks #3 and #9, then This will happen:
+
+    (A) merges #1
+    (B) merges #2
+    (C) fails
+    (D - I) are "reset" and their results are ignored by [Zuul][z]
+
+Now (C) will be reported as a failure, and [Zuul][z] will try a new scenario:
+
+    AA 4
+    BB 4+5
+    CC 4+5+6
+    DD 4+5+6+7
+    EE 4+5+6+7+8
+    FF 4+5+6+7+8+9
+
+And this happens:
+
+    (AA) merges #4
+    (BB) merges #5
+    (CC) merges #6
+    (DD) merges #7
+    (EE) merges #8
+    (FF) fails
+
+So, in the time span of _two_ test runs, we have found the two
+bad interactions, and merged 7 patches that do work well all
+together. Suddenly having hour long tests isn't such a huge problem.
+
+Meanwhile in those two test runs, new patches have only been created from
+known good working master branches, thus reinforcing good interactions
+with approved code, allowing development to progress faster and reducing
+accidental breakage. Even at the busiest time, usually just before
+releases, OpenStack sees a very long gate, and exhaustion of its 100's
+of available VM capacity across many clouds. But it still doesn't usually
+get worse than about 8 or 9 test runs to land many patches.
+
+This becomes especially important when you want to start co-gating
+projects.  In OpenStack, we don't want to merge code to [Keystone][k],
+the OpenStack Identity and Auth service, that obviously breaks [Nova][n],
+the OpenStack Compute Service, and we also don't want to land code in
+Nova that breaks [Cinder][c], the Block Storage service.
+
+These services are all developed by separate teams to cut down on
+communication overhead and build on areas of expertise. As such, they
+have well defined API's and they are able to be tested comprehensively
+in isolation. However, while having good API boundaries is extremely
+important, there are devils in the details that only show up when actual
+tests are run and code is fully integrated with dependencies.
```

Comment:
Feels like you're missing a description of co-gating, to match up with the level of description you gave gating in general. Could you add a blurb about how a change introduced to Keystone that would pass on it's own, gets blocked because it breaks something in Nova? That might help readers grasp why co-gating is powerful and important.